Libraries/dotNetRdf.Core/Writing/WriterUtilities.cs

```diff
@@ -769,6 +771,42 @@ public static string EncodeForXml(string value)
                 .Replace("\"", "&quot;");
         }
 
+        /// <summary>
+        /// Removes characters invalid in XML documents.
+        /// </summary>
+        /// <param name="content">The string to filter.</param>
+        /// <returns>
+        /// New string that does not contain any invalid characters.
+        /// </returns>
+        public static string RemoveInvalidXmlChars(string content)
```

Comment:
I wonder if it would be easier to just filter on the XML definition of a Char which is:
```
#x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]
```
and then drop characters that are not in one of those ranges. It would avoid having to handle multiple exceptions in those corner cases where a comment has lots of disallowed chars.

Reply:
Yeah, I think that would be a viable approach and would make it clear what sanitisation the formatter is doing.